### PR TITLE
LibGfx: Introduce provisional font interface

### PR DESCRIPTION
**Pull request by sunverwerth**
Old font functionality has been moved into BitmapFont and an abstract Font interface has been introduced to faciliate further development of TTF font integration.

This is mainly done to prevent merge conflicts while working on the new fonst system and someone makes changes to the old Font class.

At the moment the interface is just a reflection of the old functionality but will be refined over time as ttf integration continues.

### CODE REVIEW

File: Applications/FontEditor/main.cpp [hunk 1]
```diff
@@ -60,12 +60,12 @@ int main(int argc, char** argv)
     args_parser.add_positional_argument(path, "The font file for editing.", "file", Core::ArgsParser::Required::No);
     args_parser.parse(argc, argv);
 
-    RefPtr<Gfx::Font> edited_font;
+    RefPtr<Gfx::BitmapFont> edited_font;
     if (path == nullptr) {
         path = "/tmp/saved.font";
-        edited_font = Gfx::FontDatabase::default_font().clone();
+        edited_font = static_cast<Gfx::BitmapFont*>(&Gfx::FontDatabase::default_font().clone().leak_ref());
```
Comment (sunverwerth) [inline Applications/FontEditor/main.cpp:66]: Is there a nicer way to do this?

Comment (alimpfard) [reply Applications/FontEditor/main.cpp:66]: Make `clone()` return a mutable reference instead maybe?
const_cast is fine too though imho

**Comment by awesomekling** [comment]
Nitpick: LibGFX => LibGfx

File: Applications/FontEditor/main.cpp (hunk 1 above)
Comment (sunverwerth) [reply Applications/FontEditor/main.cpp:66]: Can you give an example? Maybe I'm a bit daft.
clone() returns a `NonnullRefPtr<Font>` but I want to convert it to a `RefPtr<BitmapFont>`

Comment (awesomekling) [reply Applications/FontEditor/main.cpp:66]: Can you use `static_ptr_cast` for this?

Comment (sunverwerth) [reply Applications/FontEditor/main.cpp:66]: Indeed, I can. Didn't know that was a thing :+1:

Comment (alimpfard) [reply Applications/FontEditor/main.cpp]: Wow, I somehow misread the entire thing here :/

Comment (sunverwerth) [reply Applications/FontEditor/main.cpp]: Happens to the best of us :smile:

**Comment by sunverwerth** [comment]
Should I change something here or can this be merged in? :thinking: 
Obviously the conflict needs to be fixed first.

**Comment by awesomekling** [comment]
This seems fine to me :)

**Comment by sunverwerth** [comment]
whew ok :smile:

**Comment by sunverwerth** [comment]
Thanks, happy new year :partying_face:

**Pull request merged**